### DISCUSSION TITLE
Display Latency in new Lazer Update

### DISCUSSION THREAD

**Discussion by myutin** [ppy/osu#29846] [Q&A]
### Type

Performance

### Bug description

Since a couple days ago, I've noticed display latency during gameplay. This appears as my cursor lagging about 100ms behind my mouse input. Judgments and timings have no latency (god bless whoever implemented multithreading).
The issue is intermittent, but happens very often on longer maps or on sections with high object density. Task manager says my GPU usage is at 100% (i have an integrated graphics card).
I tried downgrading to 2024.906.1 and that did not fix the issue. I then downgraded to 2024.817.0 and that fixed the issue for me.

### Version

2024.906.2-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16923609/compressed-logs.zip)

**Comment by myutin** [comment]
I tried changing the threading mode to single-thread in 2024.906.2. Unfortunately, this did not fix the issue for me. Glad to hear something works for you though.

**Comment by peppy** [comment]
It sounds like the game is not running in full screen anymore. I know there was an issue with [AMD drivers](https://github.com/ppy/osu/discussions/29647) recently, but not sure since it seems you guys are on intel.

**Comment by smoogipoo** [comment]
It could also be that they may need to update their GPU preference if they had one set, since the executable location changed.

**Comment by myutin** [comment]
The game has been running in full screen. (When I switch it to windowed, the latency further increases).
I did not have a GPU preference set for lazer. I've now tried setting it to high-performance, and there was no change. (Thanks for making me aware of this feature though!)
I'd be happy to provide any further information to help you fix this issue. Please let me know what you need.

**Comment by peppy** [comment]
@smoogipoo any ideas? seems like nothing we have control over.

**Comment by smoogipoo** [comment]
I would suggest trying the OpenGL renderer from the ingame settings.

**Comment by myutin** [comment]
I just tried changing the render engine. Worth a shot, but didn't fix the issue.
However, I have noticed that the latency is worst during kiai time, and mostly fine outside kiai time. (My skin removes kiai stars, so I didn't realize this sooner)

**Reply by peppy** [reply]
Star fountains are very optimised, though.

**Comment by smoogipoo** [comment]
Are you able to record a slo-mo video of this on your phone?

**Comment by myutin** [comment]
Here's a clip of Auto playing a map for 30 seconds, then playing a modified version of the same map for 30 more seconds. My phone can't seem to do slo-mo, and github uploads are limited to 10MB, so this is the best I could come up with.
(Sorry for the eight frozen seconds at the start of the video; I really need to find a better video editor).

https://github.com/user-attachments/assets/6b9b5ca3-f92d-4e9b-8a36-37cf019295d8

Here's the same events recorded in OBS. The first half of the video is extremely choppy, though the game appears smoother IRL. (osu is probably taking all of OBS's gpu time or something).

https://github.com/user-attachments/assets/a18dfa9a-cfe2-4dac-b246-ea854a033a5c

Neither of these perspectives convey what I'm seeing in gameplay, so I'll also describe it in words. Osu runs smoothly (rarely drops frames, never sustains low framerates), but seems to lag behind by 10-ish frames anytime kiai starts.

The original map is [this](https://osu.ppy.sh/beatmapsets/1474871#osu/3082637). In the modified version I removed all the kiai from the first three minutes. For reference, here's how much kiai there was in the clips above:

![image](https://github.com/user-attachments/assets/7d4dff24-f879-43af-be29-1e85c0bdf048)
